internal/provider/provider.go

```diff
@@ -146,12 +156,33 @@ func (p *PrefectProvider) Configure(ctx context.Context, req provider.ConfigureR
 		return
 	}
 
+	// Determine which profile and profile file to use
+	var profileName string
+	if !config.Profile.IsNull() {
+		profileName = config.Profile.ValueString()
+	}
+
+	var profileFilePath string
+	if !config.ProfileFile.IsNull() {
+		profileFilePath = config.ProfileFile.ValueString()
+	}
+
+	// Load profile authentication as fallback
+	profileAuth, err := LoadProfileAuth(ctx, profileName, profileFilePath)
+	if err != nil {
+		helpers.AddProfileWarning(resp, profileName, profileFilePath, err)
+		// Set profileAuth to an empty model if error occurred
+		profileAuth = &PrefectProviderModel{}
+	}
+
 	// Extract endpoint from configuration or environment variable.
 	var endpoint string
 	if !config.Endpoint.IsNull() {
 		endpoint = config.Endpoint.ValueString()
 	} else if apiURLEnvVar, ok := os.LookupEnv(envAPIURL); ok {
 		endpoint = apiURLEnvVar
+	} else if !profileAuth.Endpoint.IsNull() {
+		endpoint = profileAuth.Endpoint.ValueString()
```

Comment:
Now that we have two implicit options for configuration (env vars and profiles file), what do you think of adding a diagnostic somewhere in the output for the user to confirm which method was used? I'm picturing a situation where they have env vars set but also have a profiles file, and want to be sure that configuration is being pulled from the env vars (given that they're higher precedence).

Reply:
I think it'd be good, even if it's only a debug level log - especially since this is implicitly reading configuration without any direct user guidance